### PR TITLE
Add API tests for ACSes

### PR DESCRIPTION
**Pull request by vsedmik**
This PR adds CRUD and bulk_action tests for ACSes and uses one common fixture for both, CLI and API tests.

Requires [nailgun #911](https://github.com/SatelliteQE/nailgun/pull/911)

### CODE REVIEW

**Comment by vsedmik** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_acs.py tests/foreman/cli/test_acs.py
nailgun: 911

File: tests/foreman/api/test_acs.py
Comment (adarshdubey-star) [inline tests/foreman/api/test_acs.py:1]: Can we add `e2e` markers to any of these tests? I feel this is a good candidate `test_positive_CRUD_all_types`

**Review by Griffin-Sullivan — APPROVED** [review]
Looks good! One comment to consider. Do we need to run PRT again?

File: tests/foreman/cli/test_acs.py [hunk 1]
```diff
@@ -34,26 +33,12 @@
...
-    module_target_sat.cli.Repository.synchronize({'id': file_repo['id']})
-    yum_product = module_target_sat.cli_factory.make_product({'organization-id': module_org.id})
-    yum_repo = module_target_sat.cli_factory.make_repository(
-        {'product-id': yum_product['id'], 'content-type': 'yum', 'url': CUSTOM_RPM_REPO}
-    )
-    module_target_sat.cli.Repository.synchronize({'id': yum_repo['id']})
-    return {'yum': yum_repo, 'file': file_repo}
-
-
+@pytest.mark.e2e
```
[the remark this answers is not in the document]
Comment (vsedmik) [reply tests/foreman/cli/test_acs.py:36]: I agree.

**Review by sambible — APPROVED** [review]
Good model to base future e2e test off of - nice work

**Review by pondrejk — APPROVED** [review]

**Review by jyejare — APPROVED** [review]
ACK with non-blocking improvement comments!

File: pytest_fixtures/component/acs.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Alternate Content Sources fixtures
+import pytest
+
+from robottelo.constants.repos import CUSTOM_FILE_REPO
+from robottelo.constants.repos import CUSTOM_RPM_REPO
+
+
+@pytest.fixture(scope='module')
+def module_yum_file_repos(module_target_sat, module_org):
```
[the remark this answers is not in the document]
Comment (jyejare) [reply pytest_fixtures/component/acs.py:9]: I understood the reason why you have done it in this fashion as your test demands, but lets see if we can use them in split formatting

File: pytest_fixtures/component/acs.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    file_product = module_target_sat.api.Product(organization=module_org).create()
```
[the remark this answers is not in the document]
Comment (vsedmik) [reply pytest_fixtures/component/acs.py:10]: Here I wanted to have two different products for each type of repository, so it was rather intention not to use one common `module_product`.

File: pytest_fixtures/component/acs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vsedmik) [reply pytest_fixtures/component/acs.py:9]: Hmm, maybe I could go with something like this, if I cut the fixture into two:
```
params.update(
    {'product_ids': [module_yum_repo.product.id]}
    if cnt_type == 'yum'
    else {'product_ids': [module_file_repo.product.id]}
)
```

**Comment by vsedmik** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_acs.py tests/foreman/cli/test_acs.py
nailgun: 911

**Pull request merged**